server/chat-commands/room-settings.ts

```diff
@@ -290,7 +292,55 @@ export const commands: ChatCommands = {
 		`/slowchat [number] - Sets a limit on how often users in the room can send messages, between 2 and 60 seconds. Requires @ # &`,
 		`/slowchat off - Disables slowchat in the room. Requires @ # &`,
 	],
+	permission: 'permissions',
+	permissions: {
+		set(target, room, user) {
+			let [perm, rank] = target.split(',').map(item => item.toLowerCase().replace(/ +/g, ''));
+			perm = toID(perm);
+			if (!room.auth.atLeast(user, '#')) {
+				return this.errorReply(`/permissions set - Access denied.`);
+			}
+			if (!target || !perm) return this.parse(`/permissions help`);
+			if (rank && !Auth.isValidSymbol(rank)) {
+				return this.errorReply(`${rank} is not a valid rank.`);
+			}
+			if (Permissions.getPermissions(room)[perm] === rank) {
+				return this.errorReply(`${perm} is already set to ${rank}.`);
+			}
+			const approved = Permissions.approvedPermissions(room);
+			if (!approved.includes(perm)) return this.errorReply(`${perm} is not a valid permission.`);
+			if (!room.persist) return this.errorReply(`This room does not allow customizing permissions.`);
+			if (!Permissions.setPermission(perm, rank as GroupSymbol, room)) {
+				return this.errorReply(`${perm} cannot be configured. See /permissions help for configurable commands.`);
+			}
+			if (!rank) rank = `default`;
+			this.modlog(`SETPERMISSION`, null, `${perm}: ${rank}`);
+			return this.privateModAction(`(${user.name} set the required rank for ${perm} to ${rank}.)`);
+		},
+		sethelp: [
+			`/setpermission [command, rank] - sets the required permission to use the command [command] to [rank].`,
+			`If no [rank] is given, resets the needed permission to the default. Requires: # &`,
+		],
+
+		view(target, room, user) {
+			return this.parse(`/join view-permissions-${room.roomid}`);
+		},
 
+		help: '',
+		''(target, room, user) {
+			const configGroups = [
```

Comment:
This should probably be a global at the top of the file, or even exported from `user-groups`.

Reply:
Actually, `Permissions.approvedPermissions` already exists. Why is this separated from that?

Reply:
this is separated so it can be made clear that those permissions serve to group permissions, as opposed to configuring just one command.

Reply:
I wanted it to be clear that say, `gamemanagement` will change multiple things, as opposed to just, say, `notifyrank`.

Reply:
Even if they're different in that way, they should still be in the same part of the code and named similarly, because they serve similar functions.

Reply:
It's bad to have two drastically different implementations for what are conceptually the same thing.

Reply:
I mean, the config groups _are_ in `Permissions.approvedPermssions` already, which is where they're needed, this is just using an extra array so people know which permissions do that. I don't see how they're drastically different.